robottelo/hosts.py

```diff
@@ -359,3 +366,121 @@ def patch_os_release_version(self, distro=DISTRO_RHEL7):
+        super().__init__(*args, **kwargs)
+        self._init_nailgun()
+        self._init_cli()
+        self._init_airgun()
+
+    def _init_nailgun(self):
+        """Import all nailgun entities and wrap them under self.api"""
+        from nailgun.config import ServerConfig
+        from nailgun.entity_mixins import Entity
+        from nailgun import entities
+
+        def inject_config(cls, server_config):
+            """inject a nailgun server config into the init of nailgun entity classes"""
+            import functools
+
+            class DecClass(cls):
+                __init__ = functools.partialmethod(cls.__init__, server_config=server_config)
+
+            return DecClass
+
+        # set the server configuration to point to this satellite
+        self.nailgun_cfg = ServerConfig(
+            auth=(settings.server.admin_username, settings.server.admin_password),
+            url=f'https://{self.hostname}',
+            verify=False,
+        )
+        # add each nailgun entity to self.api, injecting our server config
+        self.api = lambda: None
+        for name, obj in entities.__dict__.items():
+            try:
+                if Entity in obj.mro():
+                    setattr(self.api, name, inject_config(obj, self.nailgun_cfg))
+            except AttributeError:
+                # not everything has an mro method, we don't care about them
+                pass
+
+    def _init_cli(self):
+        """Import all robottelo cli entities and wrap them under self.cli"""
+        import importlib
+        from robottelo.cli.base import Base
+
+        self.cli = lambda: None
+        for file in Path('robottelo/cli/').iterdir():
+            if file.suffix == '.py' and not file.name.startswith('_'):
+                cli_module = importlib.import_module(f'robottelo.cli.{file.stem}')
+                for name, obj in cli_module.__dict__.items():
+                    try:
+                        if Base in obj.mro():
+                            # set our hostname as a class attribute
+                            obj.hostname = self.hostname
+                            setattr(self.cli, name, obj)
+                    except AttributeError:
+                        # not everything has an mro method, we don't care about them
+                        pass
+
+    def _init_airgun(self):
+        """Initialize an airgun Session object and store it as self.ui_session"""
+        from airgun.session import Session
+
+        def get_caller():
+            import inspect
+
+            for frame in inspect.stack():
+                if frame.function.startswith('test_'):
+                    return frame.function
+
+        self.ui_session = Session(
+            session_name=get_caller(),
+            user=settings.server.admin_username,
+            password=settings.server.admin_password,
+            hostname=self.hostname,
+        )
+
+    def capsule_certs_generate(self, capsule_fqdn, **extra_kwargs):
```

Comment:
Consider taking the capsule instance as an arg, and referencing the relevant attribute, instead of making the caller resolve the attribute.